vc_vidyo/indico_vc_vidyo/api/client.py

```diff
@@ -77,11 +77,28 @@ def create_room_object(self, **kwargs):
 
     @raises_api_error
     def find_room(self, extension):
+        from indico_vc_vidyo.plugin import VidyoPlugin
         filter_ = self.client.factory.create('Filter')
         filter_.query = extension
+        filter_.limit = 40
         filter_.dir = 'DESC'
 
-        return self.soap.getRooms(filter_).room
+        target_room = None
+        counter = 0
+
+        while True:
```

Comment:
What happens if `self.soap.getRooms(filter_).room` doesn't return any room for some reason? Then we'd be stuck in an infinite loop. Is this really what we want?